internal/errors.go

```diff
@@ -0,0 +1,174 @@
+	// AlreadyExists is a OnePlatform error code.
+	AlreadyExists ErrorCode = "ALREADY_EXISTS"
+
+	// ResourceExhausted is a OnePlatform error code.
+	ResourceExhausted ErrorCode = "RESOURCE_EXHAUSTED"
+
+	// Cancelled is a OnePlatform error code.
+	Cancelled ErrorCode = "CANCELLED"
+
+	// DataLoss is a OnePlatform error code.
+	DataLoss ErrorCode = "DATA_LOSS"
+
+	// Unknown is a OnePlatform error code.
+	Unknown ErrorCode = "UNKNOWN"
+
+	// Internal is a OnePlatform error code.
+	Internal ErrorCode = "INTERNAL"
+
+	// Unavailable is a OnePlatform error code.
+	Unavailable ErrorCode = "UNAVAILABLE"
+
+	// DeadlineExceeded is a OnePlatform error code.
+	DeadlineExceeded ErrorCode = "DEADLINE_EXCEEDED"
+)
+
+// FirebaseError is an error type containing an error code string.
+type FirebaseError struct {
+	ErrorCode ErrorCode
+	Code      string
+	String    string
+	Response  *http.Response
+}
+
+func (fe *FirebaseError) Error() string {
+	return fe.String
+}
+
+// HasPlatformErrorCode checks if the given error contains a specific error code.
+func HasPlatformErrorCode(err error, code ErrorCode) bool {
+	fe, ok := err.(*FirebaseError)
+	return ok && fe.ErrorCode == code
+}
+
+// HasErrorCode checks if the given error contain a specific error code.
+//
+// Deprecated.
+func HasErrorCode(err error, code string) bool {
+	fe, ok := err.(*FirebaseError)
+	return ok && fe.Code == code
+}
+
+// Error creates a new FirebaseError from the specified error code and message.
+//
+// Deprecated.
+func Error(code string, msg string) *FirebaseError {
+	return &FirebaseError{
+		Code:   code,
+		String: msg,
+	}
+}
+
+// Errorf creates a new FirebaseError from the specified error code and message.
+//
+// Deprecated.
+func Errorf(code string, msg string, args ...interface{}) *FirebaseError {
+	return Error(code, fmt.Sprintf(msg, args...))
+}
+
+var httpStatusToErrorCodes = map[int]ErrorCode{
+	http.StatusBadRequest:          InvalidArgument,
+	http.StatusUnauthorized:        Unauthenticated,
+	http.StatusForbidden:           PermissionDenied,
+	http.StatusNotFound:            NotFound,
+	http.StatusConflict:            Conflict,
+	http.StatusTooManyRequests:     ResourceExhausted,
+	http.StatusInternalServerError: Internal,
+	http.StatusServiceUnavailable:  Unavailable,
+}
+
+// NewFirebaseError creates a new error from the given HTTP response.
+func NewFirebaseError(resp *Response) *FirebaseError {
+	code, ok := httpStatusToErrorCodes[resp.Status]
+	if !ok {
+		code = Unknown
+	}
+
+	return &FirebaseError{
+		ErrorCode: code,
+		String:    fmt.Sprintf("unexpected http response with status: %d\n%s", resp.Status, string(resp.Body)),
+		Response:  resp.LowLevelResponse(),
+	}
+}
+
+// NewFirebaseErrorOnePlatform parses the response payload as a GCP error response
+// and create an error from the details extracted.
+//
+// If the response failes to parse, or otherwise doesn't provide any useful details
+// CreatePlatformError creates an error with some sensible defaults.
```

Comment:
It looks like this PR removes `CreatePlatformError()`. Is this comment still valid with those changes?

Reply:
Updated comment